src/containers/my-resources/lessons-container/LessonsContainer.tsx

```diff
@@ -55,62 +46,79 @@ const LessonsContainer = () => {
     removeColumnRules
   )
 
-  const onResponseError = useCallback(
-    (error?: ErrorResponse) => {
-      dispatch(
-        openAlert({
-          severity: snackbarVariants.error,
-          message: getErrorKey(error)
-        })
-      )
-    },
-    [dispatch]
-  )
-
-  const getMyLessons = useCallback(
-    () =>
-      ResourceService.getUsersLessons({
-        limit: itemsPerPage,
-        skip: (page - 1) * itemsPerPage,
-        sort,
-        title: searchTitle.current,
-        categories: selectedItems
-      }),
-    [page, itemsPerPage, sort, searchTitle, selectedItems]
-  )
+  const getMyLessons = useCallback(() => {
+    return ResourceService.getUsersLessonsQuery({
+      limit: itemsPerPage,
+      skip: (page - 1) * itemsPerPage,
+      sort,
+      title: searchTitle.current,
+      categories: selectedItems
+    })
+  }, [page, itemsPerPage, sort, searchTitle, selectedItems])
 
   const deleteLesson = useCallback(
     (id?: string) => ResourceService.deleteLesson(id ?? ''),
     []
   )
 
-  const { response, loading, fetchData } = useAxios<
-    ItemsWithCount<Lesson>,
-    GetResourcesParams
-  >({
-    service: getMyLessons,
-    defaultResponse: defaultResponses.itemsWithCount,
-    onResponseError
+  const {
+    data: lessons,
+    isLoading,
+    error
+  } = useQuery({
+    queryKey: [
+      'lessons',
+      page,
+      itemsPerPage,
+      sort,
+      searchTitle.current,
+      selectedItems
+    ],
+    queryFn: getMyLessons,
+    options: {
+      staleTime: Infinity
+    }
   })
 
   const onEdit = (id: string) => {
-    const resource = response.items.find((item) => item._id === id)
+    if (!lessons) {
+      return
+    }
```

Comment:
Always add an empty line after `if` statements
```suggestion
    }
    
```

Reply:
Done.